aws/sdk/build.gradle.kts

```diff
@@ -452,6 +471,50 @@ tasks.register("sdkTest") {
     finalizedBy(Cargo.CLIPPY.toString, Cargo.TEST.toString, Cargo.DOCS.toString)
 }
 
+//Tasks for generating individual Cargo.lock files
+fun Project.registerLockfileGeneration(
+    dir: File,
+    name: String,
+): String {
+    val taskName = "generate${name}Lockfile"
+    tasks.register<Exec>(taskName) {
+        workingDir(dir)
+        environment("RUSTFLAGS", "--cfg aws_sdk_unstable")
+        commandLine("cargo", "generate-lockfile")
+    }
+
+    return taskName
+}
+
+val registerAwsConfigTaskName = registerLockfileGeneration(awsConfigPath, "AwsConfig")
+val registerAwsRustRuntimeTaskName = registerLockfileGeneration(awsRustRuntimePath, "AwsRustRuntime")
+val registerRustRuntimeTaskName = registerLockfileGeneration(rustRuntimePath, "RustRuntime")
+
+tasks.register<Exec>("generateAwsSdkRustLockfile") {
+    val sdkRustPath: String =
+        properties.get("aws-sdk-rust-path") ?: throw Exception("A -Paws-sdk-rust-path argument must be specified")
+    workingDir(sdkRustPath)
+    environment("RUSTFLAGS", "--cfg aws_sdk_unstable")
+    commandLine("cargo", "generate-lockfile")
+    copy {
+        from(sdkRustPath + "/Cargo.lock")
```

Comment:
```kt
from("${sdkRustPath}/Cargo.lock")
```

Reply:
Updated in https://github.com/smithy-lang/smithy-rs/pull/3689/commits/9d84980ee2e37deadb7e45450d65c5e6bc69e07c